src/libraries/math/SafeMathUint256.sol

```diff
@@ -45,4 +44,22 @@ library SafeMathUint256 {
             return b;
         }
     }
+
+    function safeToMultiply(uint256 a, uint256 b) internal constant returns (bool) {
+        uint256 c = a * b;
+        if (a == 0 || c / a == b) {
+            return true;
+        }
+        return false;
+    }
+
+    // Float [fixed point] Operations
+    function fxpMul(uint256 a, uint256 b) internal constant returns (uint256) {
+        require(safeToMultiply(a, b));
+        return a * b / 10 ** 18;
+    }
+
+    function fxpDiv(uint256 a, uint256 b) internal constant returns (uint256) {
+        return a * 10 ** 18 / b;
```

Comment:
This needs some checks to make sure it doesn't overflow.  `a * 10^18` could overflow, resulting in an incorrect number being divided by.  Probably easiest to change this to `return div(mul(a, 10**18), b)`.